pkg/cmd/roachprod/install/cockroach.go

```diff
@@ -488,17 +464,19 @@ func (h *crdbInstallHelper) generateStartArgs(
 		}
 	}
 
+	// --join flags are unsupported/unnecessary in `cockroach start-single-node`.
 	if !h.useStartSingleNode(vers) {
-		// --join flags are unsupported/unnecessary in `cockroach
-		// start-single-node`. That aside, setting up --join flags is a bit
-		// precise. We have every node point to node 1. For clusters running
-		// <20.1, we have node 1 not point to anything (which in turn is used to
-		// trigger auto-initialization node 1). For clusters running >=20.1,
-		// node 1 also points to itself, and an explicit `cockroach init` is
-		// needed.
-		if nodes[nodeIdx] != 1 || vers.AtLeast(version.MustParse("v20.1.0")) {
-			args = append(args, fmt.Sprintf("--join=%s:%d", h.c.host(1), h.r.NodePort(h.c, 1)))
+		if idx := argExists(extraArgs, "--join"); idx == -1 {
+			// If --join is not specified, have every node point to node 1. Node 1
+			// points to itself, and an explicit `cockroach init` is needed.
+			args = append(args, fmt.Sprintf("--join=%s:%d", h.c.host(1),
+				h.r.NodePort(h.c, 1)))
 		}
+		// If --join is specified it will be appended to the args below.
+		// TODO(during review): Should we parse the supplied join flags, check if
```

Comment:
I mean, wouldn't we auto-init regardless of how the join flags are constructed, as long as --skip-init isn't specified? I'm not sure I follow.